### PR TITLE
Update README and add RELEASE notes for 23.06

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Release Notes for 2.35.0
+
+## New Freatures and Improvements
+
+* Support for [KIND\_MODEL instance
```
Comment (mc-nv) [inline RELEASE.md:33]: Links must start on a new line, it's need to be applied to the all text below.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  type](https://github.com/triton-inference-server/pytorch_backend/tree/r23.06#model-instance-group-kind)
+has been extended to the PyTorch backend.
+
+* The gRPC clients can now indicate whether they want to receive the flags
+associated with each response. This can help the clients to [programmatically
+determine](https://github.com/triton-inference-server/server/blob/r23.06/docs/user_guide/decoupled_models.md#knowing-when-a-decoupled-inference-request-is-complete)
```
Comment (mc-nv) [inline RELEASE.md:39]: Indentation needs to be kept.

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  when all the responses for a given request have been received on the client 
+  side for decoupled models.
+
+* Added beta support for using
+  [Redis](https://github.com/triton-inference-server/redis_cache/tree/r23.06) as 
+  a cache for inference requests.
+
+* The 
+  [statistics extension](https://github.com/triton-inference-server/server/blob/r23.06/docs/protocol/extension_statistics.md) 
+  now includes the memory usage of the loaded models This statistics is
```
Comment (tanmayv25) [inline RELEASE.md:49]:
```suggestion
  now includes the memory usage of the loaded models. This statistics is 
```

**Comment by tanmayv25** [comment]
nit.. Otherwise looks good to me.

**Review by mc-nv — APPROVED** [review]

**Pull request merged**